docker/test/integration/MiNiFi_integration_test_driver.py

```diff
@@ -118,57 +118,57 @@ def rm_out_child(self):
-        self.check_output(timeout_seconds, output_validator, 1)
+        self.__check_output(timeout_seconds, output_validator, 1)
 
     def check_for_single_json_file_with_content_generated(self, content, timeout_seconds):
         output_validator = SingleJSONFileOutputValidator(content)
         output_validator.set_output_dir(self.file_system_observer.get_output_dir())
-        self.check_output(timeout_seconds, output_validator, 1)
+        self.__check_output(timeout_seconds, output_validator, 1)
 
     def check_for_multiple_files_generated(self, file_count, timeout_seconds, expected_content=[]):
         output_validator = MultiFileOutputValidator(file_count, [decode_escaped_str(content) for content in expected_content])
         output_validator.set_output_dir(self.file_system_observer.get_output_dir())
-        self.check_output(timeout_seconds, output_validator, file_count)
+        self.__check_output(timeout_seconds, output_validator, file_count)
 
     def check_for_at_least_one_file_with_content_generated(self, content, timeout_seconds):
         output_validator = SingleOrMultiFileOutputValidator(decode_escaped_str(content))
         output_validator.set_output_dir(self.file_system_observer.get_output_dir())
-        expected_number_of_files = timeout_seconds
-        self.check_output(timeout_seconds, output_validator, expected_number_of_files)
+        self.__check_output(timeout_seconds, output_validator)
 
     def check_for_num_files_generated(self, num_flowfiles, timeout_seconds):
         output_validator = NoContentCheckFileNumberValidator(num_flowfiles)
         output_validator.set_output_dir(self.file_system_observer.get_output_dir())
-        self.check_output(timeout_seconds, output_validator, max(1, num_flowfiles))
+        self.__check_output(timeout_seconds, output_validator, max(1, num_flowfiles))
 
-    def check_for_num_file_range_generated(self, min_files, max_files, timeout_seconds):
+    def check_for_num_file_range_generated(self, min_files, max_files, wait_time_in_seconds):
         output_validator = NumFileRangeValidator(min_files, max_files)
         output_validator.set_output_dir(self.file_system_observer.get_output_dir())
-        self.check_output_force_wait(timeout_seconds, output_validator)
+        self.__check_output_after_time_period(wait_time_in_seconds, output_validator)
 
     def check_for_an_empty_file_generated(self, timeout_seconds):
         output_validator = EmptyFilesOutPutValidator()
         output_validator.set_output_dir(self.file_system_observer.get_output_dir())
-        self.check_output(timeout_seconds, output_validator, 1)
+        self.__check_output(timeout_seconds, output_validator, 1)
 
-    def check_output_force_wait(self, timeout_seconds, output_validator):
-        time.sleep(timeout_seconds)
-        self.validate(output_validator)
+    def __check_output_after_time_period(self, wait_time_in_seconds, output_validator):
```

Comment:
Did you mean to make these "private"? According to some random internet sources, a double underscore prefix makes the interpreter use name mangling to avoid subclasses overriding these, but I see no subclasses here. The usual way to say "private" is a single underscore prefix, which seems more appropriate here, but let me know if I'm wrong.

Reply:
I usually use the double underscore for making a member "private" because of the mangling to make it harder for a user of the class to call it. I was not aware that it is only used for subclasses not to override these. In my understanding it was usually to use double underscore for private, single for protected and none for public members and I would default to using "private" members if no subclasses are there to use them.

Reply:
Okay, this sounds reasonable. Thanks for clarifying.